### PR TITLE
DNS v2: Zone Transfer Request/Accept

### PR DESCRIPTION
**Pull request by GlaciErrDev**
For #2040

https://github.com/openstack/designate/blob/056ceb7f4ba4a4b86fe212aa31a7506ac1b27f20/designate/api/v2/controllers/zones/tasks/transfer_accepts.py
https://github.com/openstack/designate/blob/056ceb7f4b/designate/api/v2/controllers/zones/tasks/transfer_requests.py

CRUD operations for `Zone Transfer Request` and `Zone Transfer Accept`

### CODE REVIEW

**Comment by jtopjian** [comment]
@GlaciErrDev Thank you for submitting this. Is this ready for a review or are you still working on it?

At first glance, it looks like we might need some additional code references. For example, where are the fields in the `TransferAccept` struct defined (how do we know that `zone_transfer_request_id` is a valid field?), and how do we know that `target_project_id` is a valid field of the Transfer Request `CreateOpts`?

Once we're able to fully confirm the fields used in the request and result structs, a review of this PR should be fairly easy. 

Thanks again for your work and let me know if you have any questions.

**Comment by GlaciErrDev** [comment]
@jtopjian Thank you for your review and comments.

Yes, it is a completely working API layer. Tested it on the deployed setup.

Left here some additional reference links ->

API Zone Transfer Request:
https://docs.openstack.org/api-ref/dns/#zone-ownership-transfers-requests

Reference to `target_project_id` field -> https://github.com/openstack/designate/blob/056ceb7f4b/designate/objects/adapters/api_v2/zone_transfer_request.py#L37 

API Zone Transfer Accept:
https://docs.openstack.org/api-ref/dns/#zone-ownership-transfers-accepts

Reference to `zone_transfer_request_id` field -> https://github.com/openstack/designate/blob/056ceb7f4b/designate/objects/adapters/api_v2/zone_transfer_accept.py#L26

If these will need to be added to the Issue description, just mention it :)

**Review by jtopjian — CHANGES_REQUESTED** [review]
@GlaciErrDev Thanks for the additional information.

I've made some comments for your review. The majority of them are just spacing issues related to the godoc code. They're nits, but semi-important as they get rendered as formatted code on godoc.org.

The other comment is about the `version` field.

One thing to note is that the DNS acceptance tests don't run by default on here. However, I've run them in a devstack environment and, short of the missing `version` field, everything else looks good.

Please let me know if you have any questions.

File: openstack/dns/v2/transfer/request/results.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	Key string `json:"key"`
+
+	// Description for the resource.
+	Description string `json:"description"`
+
+	// Status is the status of the resource.
+	Status string `json:"status"`
+
+	// Version of the resource.
+	Version int `json:"version"`
```
[the remark this answers is not in the document]
Comment (GlaciErrDev) [reply openstack/dns/v2/transfer/request/results.go:96]: @jtopjian Thanks for the comments!

Seems that `version` leaves only in the API documentation and not in actual code... 
https://docs.openstack.org/api-ref/dns/?expanded=list-zone-transfer-requests-detail,create-zone-transfer-request-detail#id54
https://github.com/openstack/designate/blob/056ceb7f4ba4a4b86fe212aa31a7506ac1b27f20/api-ref/source/parameters.yaml#L672

So I will remove it from fields and from fixtures.

**Comment by GlaciErrDev** [comment]
@jtopjian Thanks for the comments!

Seems that version leaves only in the API documentation and not in actual code...
https://docs.openstack.org/api-ref/dns/?expanded=list-zone-transfer-requests-detail,create-zone-transfer-request-detail#id54
https://github.com/openstack/designate/blob/056ceb7f4ba4a4b86fe212aa31a7506ac1b27f20/api-ref/source/parameters.yaml#L672

So I will remove it from fields and from fixtures.

File: openstack/dns/v2/transfer/request/doc.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	}
+
+	for _, transferRequest := range allTransferRequests {
+		fmt.Printf("%+v\n", transferRequest)
+	}
+
+Example to Create a Zone Transfer Request
+
+	zoneID := "99d10f68-5623-4491-91a0-6daafa32b60e"
+        targetProjectID := "f977bd7c-6485-4385-b04f-b5af0d186fcc"
```
[the remark this answers is not in the document]
Comment (GlaciErrDev) [reply openstack/dns/v2/transfer/request/doc.go:24]: It's strange, but the spacing is ok in the raw file. Maybe some issues with rendering in the diff?

**Comment by GlaciErrDev** [comment]
@jtopjian I left a comment about spacing issues.

In the raw files, spacing is Ok. Maybe there is some strange rendering behavior in diffs?

File: openstack/dns/v2/transfer/request/doc.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (GlaciErrDev) [reply openstack/dns/v2/transfer/request/doc.go:24]: It's strange, but the spacing is ok in the raw file. Maybe some issues with rendering in the diff?

File: openstack/dns/v2/transfer/request/doc.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	createOpts := transferRequests.CreateOpts{
+                TargetProjectID: targetProjectID,
```
[the remark this answers is not in the document]
Comment (GlaciErrDev) [reply openstack/dns/v2/transfer/request/doc.go:26]: It's strange, but the spacing is ok in the raw file. Maybe some issues with rendering in the diff?

File: openstack/dns/v2/transfer/accept/doc.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
+/*
+Package zones provides information and interaction with the zone API
+resource for the OpenStack DNS service.
+
+Example to List Zone Transfer Accepts
+
+        // Optionaly you can provide Status as query parameter for filtering the result.
```
[the remark this answers is not in the document]
Comment (GlaciErrDev) [reply openstack/dns/v2/transfer/accept/doc.go:7]: It's strange, but the spacing is ok in the raw file. Maybe some issues with rendering in the diff?

File: openstack/dns/v2/transfer/accept/doc.go [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	}
+
+	for _, transferAccept := range allTransferAccepts {
+		fmt.Printf("%+v\n", transferAccept)
+	}
+
+Example to Create a Zone Transfer Accept
+
+	zoneTransferRequestID := "99d10f68-5623-4491-91a0-6daafa32b60e"
+        key := "JKHGD2F7"
```
[the remark this answers is not in the document]
Comment (GlaciErrDev) [reply openstack/dns/v2/transfer/accept/doc.go:25]: It's strange, but the spacing is ok in the raw file. Maybe some issues with rendering in the diff?

File: openstack/dns/v2/transfer/accept/doc.go [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+	createOpts := transferAccepts.CreateOpts{
+                ZoneTransferRequestID: zoneTransferRequestID,
+		Key: key,
```
[the remark this answers is not in the document]
Comment (GlaciErrDev) [reply openstack/dns/v2/transfer/accept/doc.go:28]: It's strange, but the spacing is ok in the raw file. Maybe some issues with rendering in the diff?

File: openstack/dns/v2/transfer/request/results.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (GlaciErrDev) [reply openstack/dns/v2/transfer/request/results.go]: Removed it with new commit

**Comment by jtopjian** [comment]
@GlaciErrDev Please see the attached image. This is what I see when I open the file in `vim`. There's a mix of tabs and spaces which are causing the misalignment.

Normally `go fmt` would resolve this, but since these blocks of code are embedded in comments, `go fmt` ignores them. If you normalize the file to either use tabs or spaces for the code blocks, it should all be good after that.

You should also be able to see the misalignment if you run:

```shell
$ go doc openstack/dns/v2/transfer/accept
```

Screen Shot 2020-11-06 at 8 47 48 AM

**Review by jtopjian — APPROVED** [review]
LGTM - thank you!

There's still one small spacing issue, but I'll fix it post-commit :)

Thank you for your work and for contributing this.

**Pull request merged**